packages/vite/src/node/build.ts

```diff
@@ -359,6 +359,36 @@ export function resolveBuildOptions(
         : defaultModulePreload
   }
 
+  // Auto-resolve dependencies if not explicitly specified
+  if (raw && raw.lib !== false && raw.rollupOptions?.external == null) {
+    const formats = raw.lib?.formats ?? []
+    // If umd or iife is specified, we need to bundle all dependencies and cannot use external automatically
+    const isUMD = formats.includes('umd')
+    const isIIFE = formats.includes('iife')
+    if (!isUMD && !isIIFE) {
+      const configRoot = process.cwd()
+      const pkg = lookupFile(configRoot, ['package.json'])
```

Comment:
Look up `package.json` from lib entry should be safer than configRoot

Reply:
I now use `libEntry` if it is of type `string`
But lib.entry can be `InputOption` which also includes non-strings but objects.
Do I need to handle them also? Or can we expect that in such a case, the user already has a so complex config that they need to configure it manually?

Reply:
How about:

If all of the entry didn't point to the same `package.json`, disable this feature and output an warning saying that it's disabled.